plugin/src/cli_utils/mod.rs

```diff
@@ -0,0 +1,129 @@
+use kubectl_plugin::resources;
+use plugin::ExecuteOperation;
+use clap::Parser;
+use std::ops::Deref;
+pub(crate) mod localpvlvm;
+pub(crate) mod localpvzfs;
+pub(crate) mod localpvhostpath;
+pub(crate) mod clusterinfo;
+
+#[derive(Parser, Debug)]
+#[group(skip)]
+pub(crate) struct CliArgs {
+    #[clap(flatten)]
+    args: resources::CliArgs
+}
+
+impl Deref for CliArgs {
+    type Target = plugin::CliArgs;
+
+    fn deref(&self) -> &Self::Target {
+        &self.args
+    }
+}
+
+/// Storage engines supported
+#[derive(Parser, Debug)]
+pub(crate) enum Operations {
+    /// Mayastor specific commands
+    #[clap(subcommand)]
+    Mayastor(resources::Operations),
+    /// localpv-lvm specific commands
+    #[clap(subcommand)]
+    LocalpvLvm(localpvlvm::Operations),
+    /// localpv-zfs specific commands
+    #[clap(subcommand)]
+    LocalpvZfs(localpvzfs::Operations),
+    /// localpv-hostpath specific commands
+    #[clap(subcommand)]
+    LocalpvHostpath(localpvhostpath::Operations),
+    /// localpv-hostpath specific commands
+    #[clap(subcommand)]
+    ClusterInfo(clusterinfo::Operations),
+}
+
+#[async_trait::async_trait(?Send)]
+impl ExecuteOperation for Operations {
+    type Args = CliArgs;
+    type Error = Error;
+
+    async fn execute(&self, cli_args: &CliArgs) -> Result<(), Error> {
+        match self {
+            Operations::Mayastor(maya_ops) => {
+                resources::init_rest(&cli_args.args).await?;
+                maya_ops.execute(&cli_args.args).await?;
+            },
+            Operations::LocalpvLvm(lvm_ops) => {
+                lvm_ops.execute(&cli_args).await?;
+            },
+            Operations::LocalpvZfs(zfs_ops) => {
+                zfs_ops.execute(&cli_args).await?;
+            },
+            Operations::LocalpvHostpath(hostpath_ops) => {
+                hostpath_ops.execute(&cli_args).await?;
+            },
+            Operations::ClusterInfo(cluster_info) => {
+                cluster_info.execute(&cli_args).await?;
+            }
+        }
+        Ok(())
+    }
+}
+
+/// Wrapper error for all stem commnds.
+pub(crate) enum Error {
+    Upgrade(upgrade::error::Error),
```

Comment:
Suggest you separate the stem errors as well, and then you can handle the restplugin errors just for mayastor

Reply:
Its separated already, `enum Error ` is defined in every localpv module. Without any implementation it just has `Generic` variant. We will add more when we implement.

Reply:
I mean with stems, example:
```
Error {
Mayastor( MayastorError),
Lvm( LvmError)
}
```
Then you can hide specific errors entirely on the respective modules, rather than bring them up to the stem